### PR TITLE
CORS-2291: Create new terraform log file

### PR DESCRIPTION
**Pull request by rna-afk**
Creating a new file for terraform logging and passing all TRACE logs to a specific file.

### CODE REVIEW

**Comment by r4f4** [comment]
/remove-lifecycle stale
We're just waiting on https://github.com/openshift/installer/pull/6532 so that the terraform logs are properly piped to files in the installer side.

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 1]
```diff
@@ -338,7 +338,15 @@ if [ ! -z "${OPENSHIFT_INSTALL_PROMTAIL_ON_BOOTSTRAP:-}" ]; then
...
-TF_LOG=debug openshift-install --dir="${dir}" create cluster 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
+TF_LOG_PATH="${dir}/terraform.txt"
+TF_LOG_CORE="DEBUG"
+TF_LOG_PROVIDER="DEBUG"
+
+export TF_LOG_PATH
+export TF_LOG_CORE
+export TF_LOG_PROVIDER
+
+openshift-install --dir="${dir}" create cluster 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
```
Comment (patrickdillon) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:349]: 1. I think we need to compress the TF log output
2. We are scrubbing sensitive data here. Perhaps we want to scrub and compress in `populate_artifact_dir()`

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 2]
```diff
@@ -14,6 +14,12 @@ function populate_artifact_dir() {
     s/X-Auth-Token.*/X-Auth-Token REDACTED/;
     s/UserData:.*,/UserData: REDACTED,/;
     ' "${dir}/.openshift_install.log" > "${ARTIFACT_DIR}/.openshift_install.log"
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${dir}/terraform.log" > "${dir}/terraform.log"  
+  tar -czvf "${dir}/terraform.tar.gz" "${dir}/terraform.txt"
```
Comment (r4f4) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:22]:
```suggestion
  tar -czvf "${dir}/terraform.tar.gz" --remove-files "${dir}/terraform.txt" 
```

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 3]
```diff
@@ -14,6 +14,12 @@ function populate_artifact_dir() {
     s/X-Auth-Token.*/X-Auth-Token REDACTED/;
     s/UserData:.*,/UserData: REDACTED,/;
     ' "${dir}/.openshift_install.log" > "${ARTIFACT_DIR}/.openshift_install.log"
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${dir}/terraform.log" > "${dir}/terraform.log"
```
Comment (r4f4) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:21]:
```suggestion
    ' "${dir}/terraform.txt" > "${dir}/terraform.txt"  
```

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh [hunk 4]
```diff
@@ -14,6 +14,12 @@ function populate_artifact_dir() {
     s/X-Auth-Token.*/X-Auth-Token REDACTED/;
     s/UserData:.*,/UserData: REDACTED,/;
     ' "${dir}/.openshift_install.log" > "${ARTIFACT_DIR}/.openshift_install.log"
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${dir}/terraform.log" > "${dir}/terraform.log"  
+  tar -czvf "${dir}/terraform.tar.gz" --remove-files "${dir}/terraform.txt"
```
Comment (r4f4) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:22]:
```suggestion
  tar -czvf "${ARTIFACT_DIR}/terraform.tar.gz" --remove-files "${dir}/terraform.txt" 
```
So it's actually saved when the artifacts dir is copied.

**Comment by patrickdillon** [comment]
We need openshift/installer#6532 to test this in a rehearsal, right?

**Comment by r4f4** [comment]
> We need [openshift/installer#6532](https://github.com/openshift/installer/pull/6532) to test this in a rehearsal, right?

@patrickdillon correct

**Comment by r4f4** [comment]
/pj-reharse refresh
I'd like to see installer jobs that are not openshift-priv

**Comment by r4f4** [comment]
/pj-rehearse ci/rehearse/openshift/installer/master/e2e-aws-ovn

**Comment by r4f4** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-shared-vpc

**Comment by rna-afk** [comment]
Seems to work. I see a 1.7MB tar file with a 30MB terraform.txt file

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/32411/rehearse-32411-pull-ci-openshift-external-dns-operator-main-e2e-azure-operator/1602331451259359232/artifacts/e2e-azure-operator/ipi-install-install/artifacts/

**Pull request merged**